### PR TITLE
Fix clicking on anchors inside summary tags

### LINKED ISSUE #91184: rustdoc: Can't click any links in methods that use `details`
For example, I can't click any links (i.e., when I click them, nothing happens) in [this method's](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_ast_lowering/struct.LoweringContext.html#method.lower_expr_try_block) header. Likewise, I can't click any links for `make_async_expr`, but I **can** click links in `wrap_in_try_constructor`, which has no `details` tag.

This same behavior occurs in the `std` docs, though every `std` method is documented and thus has a `details` tag, so there's no comparison to non-`details` methods.

This is a regression in nightly 1.58.0. I'm using Safari on macOS; I don't know if this reproduces on other browsers.

### PR DESCRIPTION
**Pull request by jsha**
Fixes #91184

r? @camelid

### CODE REVIEW

File: src/librustdoc/html/static/js/main.js [hunk 1]
```diff
@@ -888,7 +888,7 @@ function hideThemeButtonState() {
 
     onEachLazy(document.querySelectorAll(".rustdoc-toggle > summary:not(.hideme)"), function(el) {
         el.addEventListener("click", function(e) {
-            if (e.target.tagName != "SUMMARY") {
+            if (e.target.tagName != "SUMMARY" && e.target.tagName != "A") {
```
Comment (camelid) [inline src/librustdoc/html/static/js/main.js:891]: This is definitely an improvement, but it seems somewhat fragile. Are there other tags that are used as click targets?

**Comment by camelid** [comment]
We want this to get in before nightly becomes beta: @bors p=1

File: src/librustdoc/html/static/js/main.js (hunk 1 above)
Comment (jsha) [reply src/librustdoc/html/static/js/main.js:891]: Buttons and forms have a default behavior on click, but those don't show up in our summaries. You're right that this feels fragile, though. Perhaps as a followup we should flip the logic and say "preventDefault if the click target is `h3`, `h4`, etc". That seems to be the usual click target in these when it's not the anchor, but need to do a little testing to confirm.

Comment (jsha) [reply src/librustdoc/html/static/js/main.js:891]: A quick check shows we'd also have to cover `<div class="stab unstable">` and friends - the item-info divs.

Comment (camelid) [reply src/librustdoc/html/static/js/main.js:891]: > but those don't show up in our summaries.

> A quick check shows we'd also have to cover `<div class="stab unstable">` and friends - the item-info divs.

I thought item-infos don't show up in the summaries either? Or do they and I'm misunderstanding?

Comment (jsha) [reply src/librustdoc/html/static/js/main.js:891]: item-infos are currently inside the summary. The idea is that they should be shown even when the whole document is collapsed, so you can see at a glance what is deprecated, unstable, platform-specific, etc.

Those and the "notable traits" hover div are the main surprising things that are inside summaries right now.

**Pull request merged**